### PR TITLE
Fix gtHashValue to properly hash all the bits when we have a 64-bit item

### PR DESCRIPTION
**Pull request by briansull**

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet/jit-contrib PTAL

File: src/jit/gentree.cpp [hunk 1]
```diff
@@ -1915,8 +1926,18 @@ unsigned Compiler::gtHashValue(GenTree* tree)
                 break;
         }
 
-        // narrowing cast, but for hashing.
-        hash = genTreeHashAdd(hash, (unsigned)add);
+        // clang-format off
```
Comment (BruceForstall) [inline src/jit/gentree.cpp]: Why do you need to turn off clang-format? seems like pretty simple formatting.

File: src/jit/gentree.cpp [hunk 2]
```diff
@@ -1915,8 +1926,18 @@ unsigned Compiler::gtHashValue(GenTree* tree)
...
+        // narrow 'add' into a 32-bit 'val'
+        unsigned val;
+#ifdef _TARGET_64BIT_
+        val = genTreeHashAdd(uhi32(add), ulo32(add));
+#else // 32-bit target
+        val = add;
+        val = add;
```
Comment (BruceForstall) [inline src/jit/gentree.cpp]: this line appears twice

File: src/jit/gentree.cpp (hunk 1 above)
Comment (briansull) [reply src/jit/gentree.cpp]: Because it gets it wrong here:  It wants me to add 9 spaces

Comment (briansull) [reply src/jit/gentree.cpp]: // clang-format off
        // narrow 'add' into a 32-bit 'val'
        unsigned val;
#ifdef _TARGET_64BIT_
        val = genTreeHashAdd(uhi32(add), ulo32(add));
#else // 32-bit target
        val = add;
#endif
        // clang-format on

File: src/jit/gentree.cpp (hunk 2 above)
Comment (briansull) [reply src/jit/gentree.cpp]: Fixed

**Pull request merged**